app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
```

Comment:
email.downcase khác gì với  email.downcase! em

Reply:
.downcase và .downcase! đều chuyển chữ in hoa về chữ in thường ạ. nhưng .downcase! sau khi chuyển về chữ in thường sẽ lưu lại giá trị được thay đổi . còn sau khi dùng .downcase thì puts nó vẫn ra chữ IN HOA như ban đầu 
E hiểu rồi ạ !